java/src/main/native/src/TableJni.cpp

```diff
@@ -2358,17 +2371,39 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_timeRangeRollingWindowAgg
     std::vector<std::unique_ptr<cudf::column>> result_columns;
     for (int i(0); i < values.size(); ++i) {
       int agg_column_index = values[i];
+      cudf::column_view const &order_by_column = input_table->column(orderbys[i]);
+      cudf::data_type order_by_type = order_by_column.type();
+      cudf::data_type unbounded_type = order_by_type;
+
+      if (cudf::is_timestamp(order_by_type)) {
```

Comment:
What do you think about using a `switch` statement for this part?

Reply:
Done